src/backends/odbc/standard-use-type.cpp

```diff
@@ -212,14 +211,20 @@ void odbc_standard_use_type_backend::pre_use(indicator const *ind)
     SQLSMALLINT sqlType(0);
     SQLSMALLINT cType(0);
     SQLLEN size(0);
+    SQLLEN bufLen(0);
 
     void* const sqlData = prepare_for_bind(size, sqlType, cType);
-
+    if (sqlType == SQL_LONGVARCHAR)
```

Comment:
Even if we want to do it like this, it doesn't make much sense to do it like this: why explicitly set `sqlType` to `SQL_LONGVARCHAR` inside `prepare_for_bind()` only to reset it to `SQL_VARCHAR` here? Instead, the code in `prepare_for_bind()` added by c025cc86 should be simply reverted.

But I am not sure if doing this would be correct, after all it wasn't added without reason and did fix #287, so I'm afraid doing this might reintroduce that bug. It looks like there might be a difference in behaviour between ODBC 10 and 11 here, has this patch (and the "long string" ODBC unit test) been tested with both 10 and 11?